### ISSUE TITLE
Loss value returned by the SFT Trainer 

### ISSUE THREAD

**Issue by SreevaatsavB** [huggingface/trl#1575]
I need some clarity regarding the loss value returned by the SFT Trainer. 

For reference, from the figure below 
<img width="1030" alt="Screenshot 2024-04-23 at 1 07 18 PM" src="https://github.com/huggingface/trl/assets/80395869/9cd015de-452a-4fc9-8013-67c994635592">

This is a LLM finetuning task, I have used Mistral-7b 4 bit model for this one. While calulating the loss value in training for 1 step, the cross entropy loss between the logits and labels is 4.77 but the loss returned from the SFT trainer is 12.02.

(I have done this by manually adding the print statements in the source code, for checking the values and understanding the flow)

Why isn't the SFT Trainer logging the loss returned from the mistral's forward pass, but some other value? What is the loss retuend by the SFT Trainer composed of then?

This is the loss returned from the base llm:- 
<img width="496" alt="Screenshot 2024-04-23 at 5 07 18 PM" src="https://github.com/huggingface/trl/assets/80395869/31715271-7bb6-4c8c-98c1-bf0bb8581739">
